Mathlib/GroupTheory/Perm/Fin.lean

```diff
@@ -306,6 +301,132 @@ end Fin
+
+theorem cycleIcc_of_gt (hij : i ≤ j) (h : j < k) : (cycleIcc i j) k = k := by
+  have kin : k ∈ Set.range ⇑(natAdd_castLEEmb (Nat.sub_le n i)) := by
+    rw [range_natAdd_castLEEmb, Set.mem_setOf_eq]
+    omega
+  have : (((j - i).castLT (sub_val_lt_sub hij)).cycleRange (((addNatEmb
+      (n - (n - i.1))).trans (finCongr _).toEmbedding).toEquivRange.symm ⟨k, kin⟩))
+      = subNat i.1 (k.cast (by omega)) (by simp [le_of_lt (lt_of_le_of_lt hij h)]) := by
+    rw [cycleIcc_simp_lemma (le_of_lt (lt_of_le_of_lt hij h)), cycleRange_of_gt]
+    exact lt_def.mpr (by rw [coe_castLT, sub_val_of_le hij, coe_subNat, coe_cast]; omega)
+  simpa only [cycleIcc_aux hij kin, natAdd_castLEEmb, this] using eq_of_val_eq (by
+    simp only [Function.Embedding.trans_apply, addNatEmb_apply, coe_toEmbedding, finCongr_apply,
+      coe_cast, coe_addNat, coe_subNat]; omega)
+
+theorem cycleIcc_of (h1 : i ≤ k) (h2 : k ≤ j) [NeZero n] :
+    (cycleIcc i j) k = if k = j then i else k + 1 := by
+  have hij : i ≤ j := le_trans h1 h2
+  have kin : k ∈ Set.range ⇑(natAdd_castLEEmb (Nat.sub_le n i)) := by
+    rw [range_natAdd_castLEEmb, Set.mem_setOf_eq]
+    omega
+  simp only [cycleIcc_aux hij kin, natAdd_castLEEmb, cycleIcc_simp_lemma h1,
+    Function.Embedding.trans_apply, addNatEmb_apply, coe_toEmbedding, finCongr_apply]
+  refine eq_of_val_eq ?_
+  split_ifs with h3
+  · have h : subNat i.1 (j.cast (by omega)) (by simp [hij]) = (j - i).castLT (sub_val_lt_sub hij) :=
+      eq_of_val_eq (by simp [subNat, coe_castLT, sub_val_of_le hij])
+    simpa [h3, cycleRange_of_eq h] using by omega
+  · have h : subNat i.1 (k.cast (by omega)) (by simp [h1]) < (j - i).castLT (sub_val_lt_sub hij) :=
+      by simp only [subNat, coe_cast, lt_iff_val_lt_val, coe_castLT, sub_val_of_le hij]; omega
+    rw [cycleRange_of_lt h, subNat]
+    simp only [coe_cast, add_def, val_one', Nat.add_mod_mod, addNat_mk, cast_mk]
+    rw [Nat.mod_eq_of_lt (by omega), Nat.mod_eq_of_lt (by omega)]
+    omega
+
+theorem cycleIcc_of_ge_and_lt (h1 : i ≤ k) (h2 : k < j) [NeZero n] : (cycleIcc i j) k = k + 1 := by
+  simp [cycleIcc_of h1 (le_of_lt h2), Fin.ne_of_lt h2]
+
+@[simp]
+theorem cycleIcc_of_last (hij : i ≤ j) [NeZero n] : (cycleIcc i j) j = i := by
+  simp [cycleIcc_of hij (ge_of_eq rfl)]
+
+theorem cycleIcc_of_trivial (hijk : k < i ∨ j < k) : (cycleIcc i j) k = k := by
+  rcases Decidable.em (i ≤ j) with hij | hij
+  · exact Or.casesOn hijk (fun hki ↦ cycleIcc_of_lt hij hki) fun hjk ↦ cycleIcc_of_gt hij hjk
+  · simp [cycleIcc, hij]
+
+@[simp]
+theorem sign_cycleIcc (hij : i ≤ j) : Perm.sign (cycleIcc i j) = (-1) ^ (j - i : ℕ) := by
+  simp [cycleIcc, hij, sub_val_of_le hij]
+
+theorem isCycle_cycleIcc (hij : i < j) : (cycleIcc i j).IsCycle := by
+  simpa [cycleIcc, le_of_lt hij] using Equiv.Perm.IsCycle.extendDomain
+    (natAdd_castLEEmb _).toEquivRange (isCycle_cycleRange (castLT_sub_nezero hij))
+
+theorem cycleType_cycleIcc (hij : i < j) : Perm.cycleType (cycleIcc i j) = {(j - i + 1: ℕ)} := by
```

Comment:
Can you add the equivalent description of the cycleType in all cases beyond `i < j`?